common/include/pcl/experimental/executor/default/cuda_executor.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#ifdef CUDA
+  #include <cuda_runtime_api.h>
+#endif
+
+#include <pcl/experimental/executor/default/base_executor.hpp>
+#include <pcl/experimental/executor/default/inline_executor.hpp>
+
+namespace executor {
+
+#ifdef CUDA
+template <typename F>
+__global__ void global_kernel(F f) {
+  f();
+}
+#endif
+
+template <typename Blocking, typename ProtoAllocator>
+struct cuda_executor;
+
+#ifdef CUDA
+template <>
+struct is_executor_available<cuda_executor> : std::true_type {};
+#endif
+
+template <typename Blocking = blocking_t::always_t,
+          typename ProtoAllocator = std::allocator<void>>
+struct cuda_executor {
+  using shape_type = typename std::array<int, 6>;
+
+  template <typename Executor, instance_of_base<cuda_executor, Executor> = 0>
+  friend bool operator==(const cuda_executor& lhs,
+                         const Executor& rhs) noexcept {
+    return std::is_same<cuda_executor, Executor>::value;
+  }
+
+  template <typename Executor, instance_of_base<cuda_executor, Executor> = 0>
+  friend bool operator!=(const cuda_executor& lhs,
+                         const Executor& rhs) noexcept {
+    return !operator==(lhs, rhs);
+  }
+
+  template <typename F>
+  void execute(F& f) const {
+#ifdef CUDA
+    void* global_kernel_args[] = {static_cast<void*>(&f)};
+    cudaLaunchKernel(reinterpret_cast<void*>(global_kernel<F>), 1, 1,
```

Comment:
Interesting aspect: if you don't use the function's address, you're forced to use a reinterpret cast. If you do, that static_cast works (as shown in the line above this this one).